MonoGame.Framework.Content.Pipeline/Graphics/DxtTexture/Bitmap.cs

```diff
@@ -0,0 +1,320 @@
+        /// <param name="width">The width of the image.</param>
+        /// <param name="height">The height of the image.</param>
+        /// <param name="blocks">The compressed DXT blocks.</param>
+        /// <param name="flags">Compression flags.</param>
+        public static Bitmap Decompress(int width, int height, Byte[] blocks, CompressionMode flags)
+        {
+            var img = new Bitmap(new Byte[width * height * 4], width, height);
+
+            DecompressImage(img, blocks, flags);
+
+            return img;
+        }
+
+        /// <summary>
+        /// Compresses an image in memory.
+        /// </summary>
+        /// <param name="flags">Compression Flags</param>
+        /// <returns>The compressed blocks</returns>
+        /// <remarks>
+        /// The source pixels should be presented as a contiguous array of width* height
+        ///
+        /// rgba values, with each component as 1 byte each.In memory this should be:
+        ///   
+        ///       { r1, g1, b1, a1, .... , rn, gn, bn, an } for n = width* height
+        ///
+        /// The flags parameter should specify either kDxt1, kDxt3 or kDxt5 compression,
+        /// however, DXT1 will be used by default if none is specified.When using DXT1 
+        /// compression, 8 bytes of storage are required for each compressed DXT block.
+        /// DXT3 and DXT5 compression require 16 bytes of storage per block.
+        ///
+        /// The flags parameter can also specify a preferred colour compressor and
+        /// colour error metric to use when fitting the RGB components of the data. 
+        /// Possible colour compressors are: kColourClusterFit (the default), 
+        /// kColourRangeFit or kColourIterativeClusterFit.Possible colour error metrics         
+        /// are: kColourMetricPerceptual(the default) or kColourMetricUniform.If no
+        /// flags are specified in any particular category then the default will be
+        /// used.Unknown flags are ignored.
+        /// 
+        /// When using kColourClusterFit, an additional flag can be specified to
+        /// weight the colour of each pixel by its alpha value.For images that are
+        /// rendered using alpha blending, this can significantly increase the
+        /// perceived quality.
+        /// 
+        /// Internally this function calls squish::Compress for each block. To see how
+        /// much memory is required in the compressed image, use
+        /// squish::GetStorageRequirements.
+        /// </remarks>
+        public Byte[] Compress(CompressionMode flags)
+        {
+            var l = GetStorageRequirements(_Width, _Height, flags);
+
+            var blocks = new Byte[l];
+
+            CompressImage(this, blocks, flags);
+
+            return blocks;
+        }        
+        
+        private static void CompressImage(Bitmap srcImage, Byte[] blocks, CompressionMode flags)
+        {
+            // fix any bad flags
+            flags = flags.FixFlags();

```

Comment:
This is really poor...  for every block in a 2048x2048 (that would be 262,144 blocks) it fixes the flags?  This should be done outside the image loop.